sockeye/inference.py

```diff
@@ -631,11 +644,51 @@ def _make_result(self,
                                 translation=target_string,
                                 tokens=target_tokens,
                                 attention_matrix=attention_matrix,
-                                score=neg_logprob)
+                                score=translation.score)
+
+    def _concat_translations(self, translations: List[Translation]) -> Translation:
+        """
+        Combine translations through concatenation.
+
+        :param translations: A list of translations (sequence, attention_matrix), score and length.
+        :return: A concatenation if the translations with a score.
+        """
+        # Concatenation of all target ids without BOS and EOS
+        target_ids = list(itertools.chain.from_iterable(translation.target_ids[1:-1] for translation in translations))
```

Comment:
Do we guarantee that 0 and -1 will be BOS and EOS at this point regardless of model settings and think this will hold for all future models?

Reply:
that's a good point. As this is a "private" method we're somewhat save, but I agree that this is something that could easily change in the future. I'll add an assertion to fail early if we every change this behavior. The BOS is guaranteed through the way `_beam_search` is set up, see [see](https://github.com/awslabs/sockeye/blob/master/sockeye/inference.py#L733). However now that I think about it, if we reach the max length and the top hypothesis has not finished yet we might actually drop the last word. I'll update the code accordingly.

Reply:
Yes, there is no guarantee that the last word is EOS